pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -186,6 +205,28 @@ func (r *ReconcileEgressGateway) Reconcile(ctx context.Context, request reconcil
 			reqLogger.Info("EgressGateway object not found")
 			// Since the EGW resource is not found, remove the deployment.
 			r.status.RemoveDeployments(types.NamespacedName{Name: request.Name, Namespace: request.Namespace})
+
+			// In case of openShift, we are using a single SCC.
+			// Whenever a EGW resource is deleted, remove the corresponding user from the SCC
+			// and update the resource.
+			if r.provider == operatorv1.ProviderOpenShift {
+				scc, err := getOpenShiftSCC(ctx, r.client)
+				if err != nil {
+					reqLogger.Error(err, "Error querying SecurityContextConstraints")
```

Comment:
we need to return an error here. If we do, we can also get rid of the `else` branch I think